packages/iov-ethereum/src/ethereumcodec.spec.ts

```diff
@@ -168,5 +168,83 @@ describe("ethereumCodec", () => {
         otherSignatures: [],
       });
     });
+
+    it("interprets ERC20 transfer of unknown contract as ETH send", () => {
+      // There is only one transaction type on Ethereum. Smart contract interaction and
+      // ETH sends can only reliably be differentiated when all contract addresses are known.
+      // As a result, we interprete all unknown transactions as ETH send.
+
+      // curl -sS -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x80295fc8cdf6ac5fce39f34037f07d5be3abe82baa8468196faf1f00ced239e3"],"id":1}' https://rinkeby.infura.io | jq .result
+      const rawGetTransactionByHashResult: EthereumRpcTransactionResult = {
+        blockHash: "0x135592131306762eef45d8f12c3d27c5d709c84d124f0df062d0bb3806a32701",
+        blockNumber: "0x3f20f6",
+        from: "0x9bd26664827550982960b9e76bcd88c0b6791bb4",
+        gas: "0x226c8",
+        gasPrice: "0x3b9aca00",
+        hash: "0x80295fc8cdf6ac5fce39f34037f07d5be3abe82baa8468196faf1f00ced239e3",
+        input:
+          "0xa9059cbb0000000000000000000000009ea4094ed5d7e089ac846c7d66fc518bd24753ab0000000000000000000000000000000000000000000000000000000000000002",
+        nonce: "0x1",
+        r: "0xcbe96b38321e6ef536da5e74b558cf87acdda825be35be40627b2b3d8633b8f4",
+        s: "0x7fc31ca5bb3dbd02e5e8fc5093082f3f2ab3e0042d4e5b25fe09e5f7485d83b7",
+        to: "0xc778417e063141139fce010982780140aa0cd5ab",
+        transactionIndex: "0x7",
+        v: "0x2b",
+        value: "0x0",
+      };
+      const expectedPubkey = fromHex(
+        "040b8b6f82e7226d21991dd6b1a7de357cebfc42ccb95678404d8e2b54cc3be187b17a50ef833884df318aa7def070585c92a185272b8cb8b61ba916d993435c87",
+      ) as PublicKeyBytes;
+
+      const postableBytes = Encoding.toUtf8(JSON.stringify(rawGetTransactionByHashResult)) as PostableBytes;
+
+      const codec = new EthereumCodec({ erc20Tokens: new Map<TokenTicker, Erc20Options>() });
+      expect(codec.parseBytes(postableBytes, "ethereum-eip155-4" as ChainId)).toEqual({
+        transaction: {
+          kind: "bcp/send",
+          creator: {
+            chainId: "ethereum-eip155-4" as ChainId,
+            pubkey: {
+              algo: Algorithm.Secp256k1,
+              data: expectedPubkey,
+            },
+          },
+          fee: {
+            gasLimit: {
+              quantity: "141000",
+              fractionalDigits: 18,
+              tokenTicker: "ETH" as TokenTicker,
+            },
+            gasPrice: {
+              quantity: "1000000000",
+              fractionalDigits: 18,
+              tokenTicker: "ETH" as TokenTicker,
+            },
+          },
+          amount: {
+            quantity: "0",
+            fractionalDigits: 18,
+            tokenTicker: "ETH" as TokenTicker,
+          },
+          recipient: "0xc778417E063141139Fce010982780140Aa0cD5Ab" as Address,
+          // Non-UTF8 automatically represented as hex
```

Comment:
What does this mean exactly? Is this memo in UTF8 or hex?

Reply:
The thing is: it is not a memo. BCP uses Ethereum's `input` field to encode a memo as UTF-8. But the other way round we have no idea what kind of binary `input` data comes from Ethereum. So as soon as as receive `input` that is not a valid UTF-8 representation of a string, we represent that data as hex.